### PR TITLE
Comid

### PR DESCRIPTION
**Pull request by stevemessick**
Not sure what happened with the commits. I thought I had deleted a bunch of mistakes, but there they are!

This does not actually switch dart-pad to use comid. The import in codemirror_module.dart needs to be changed to do that.

### CODE REVIEW

File: lib/editing/editor_comid.dart [hunk 1]
```diff
@@ -0,0 +1,297 @@
+// Copyright (c) 2014, the Dart project authors.  Please see the AUTHORS file
```
Comment (devoncarew) [inline lib/editing/editor_comid.dart]: nit: 2015 :)

File: lib/editing/editor_comid.dart [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+
+final CodeMirrorFactory codeMirrorFactory = new CodeMirrorFactory._();
+
+final _gutterId = 'CodeMirror-lint-markers';
+
+class CodeMirrorFactory extends EditorFactory {
+
+  CodeMirrorFactory._();
+
+  List<String> get modes => null; // TODO CodeMirror.MODES; unused
```
Comment (devoncarew) [inline lib/editing/editor_comid.dart]: Is there a comid analog for these getters?

Comment (devoncarew) [inline lib/editing/editor_comid.dart]: If not, perhaps this wrapper around comid should have some hard-coded values? `dart, html, css`?

**Comment by devoncarew** [comment]
@stevemessick, 1 question, 1 nit. Can you update your PR? Looking forward to playing with comid!

File: lib/editing/editor_comid.dart (hunk 2 above)
Comment (stevemessick) [inline lib/editing/editor_comid.dart]: So far, I have not extended the CodeMirror API. If we need a list of themes and modes we can add it here, but they are currently unused.

**Comment by stevemessick** [comment]
I forgot to set user.email to messick@google.com for this repos. I have
done that. I thought I was already set up to use my personal email, though:
steve.messick@gmail.com.

**Comment by devoncarew** [comment]
lgtm! Merging...

**Pull request merged**